### ISSUE TITLE
[Elixir] Implement concept exercise: numbers

### ISSUE THREAD

**Issue by neenjaw** [exercism/v3#1488]
This issue describes how to implement the `numbers` concept exercise for the Elixir track.

[🔗 **How to help**](#actionable-items)

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This is a fork from the [Javascript numbers exercise](https://github.com/exercism/v3/blob/master/languages/javascript/exercises/concept/numbers/.docs/instructions.md) which uses the [Freelancer Rates Story](https://github.com/exercism/v3/blob/master/reference/stories/numbers.freelancer-rates.md)

The goal of this exercise is to teach how to use floating point numbers in Elixir

## Learning objectives

- Know how to recognise floating point numbers
- Know how to write floating point literals
- Know their underlying type (double precision, IEEE 754 encoding)
- Know their basic limitations
  - not all numbers can be represented, eg: (0.1 + 0.2)
- Know how to truncate floating point numbers to a certain decimal place
  - via Float.ceil, Float.floor, Float.round
- Know how to convert float to integers

## Out of scope

- multiple-function-clauses

## Concepts

- `floating-point-numbers`

## Prerequisites

- `basics`

## Narrative for exercise

Following the [Freelancer Rate Story](https://github.com/exercism/v3/blob/master/reference/stories/numbers.freelancer-rates.md), students should be tasked to create a series of functions calculating the `daily_rate`, `monthly_rate`, `days_in_budget` functions.

See: [Javascript numbers](https://github.com/exercism/v3/blob/master/languages/javascript/exercises/concept/numbers/.docs/instructions.md)

## Resources to refer to

- [Float](https://hexdocs.pm/elixir/Float.html#ceil/2)
- <http://0.30000000000000004.com/>
- [The Floating-Point Guide](https://floating-point-gui.de/)

### Hints

- slight hints to guide toward the solution without giving away exact functions/code/implmentation

### After

- what floats are
- what precision they are
- how big are they (64 bit)
- limitations
  - range of numbers which can be expressed
- truncation of float, conversion to integer

## Representer

- TBD

## Analyzer

- TBD

## Actionable items <a href="#actionable-items" id="actionable-items">🔗</a>

looking for help in a few ways, and I am here to help you through getting this exercise off the ground:

- [ ] implement the [Freelancer Rate](https://github.com/exercism/v3/blob/master/reference/stories/numbers.freelancer-rates.md) narrative to teach nil values
  - Write the exemplar solution (How can we fulfill the premise of the exercise with following the scope of requirements)
  - If we can't fulfill the premise with the current scope, then we need to adjust:
    - the premise
    - the code
    - the requirements
  - let's talk about it! We can help you!
- [ ] Writing the test cases (if TDD is your thing!)
- [ ] Writing the student facing documents
  - I am happy to do this for/with you!

For a first phase review we need:

- [ ] the [design.md document](https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md#add-metadesignmd-file) (the information in this post)
- [ ] the examplar solution
- [ ] the [introduction document](https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md#add-docsintroductionmd-file) describing the concepts to be taught

Thanks for considering!

## Need Help?

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by angelikatyborska** [comment]
I am going to work on this one. At first glance, everything's clear to me here.

**Comment by angelikatyborska** [comment]
Ok, I have some doubts about the discount.

If I follow the example and accept a discount as a string like the story suggests, e.g. `"0.0%", I have to do this:
```elixir
iex> Float.parse("0.0%")    
{0.0, "%"}
```

That means that the prerequisites for this exercise would be also `string-literals`, `tuples`, and `pattern-matching`. However, the `tuples` exercise has `floating-point-numbers` as a prerequisite and we would have a deadlock.

If I drop the "%" sign from the string, I can do this:
```
iex> String.to_float("0.0") 
0.0
```

Which no longer requires `tuples` and `pattern-matching`, only requires `string-literals`, which is acceptable. However, now it seems a bit forced to me that this argument is supposed to be a string.

I would be inclined to make the discount a float. It would lose the benefit of teaching extracting numbers from strings (which is not listed as a learning objective here anyway), but it would still keep the benefit of introducing division (if 50% is passed as 50, it needs to be divided by 100). That would simplify the exercise and make it a likely candidate to be one of the very first ones, truly requiring only `basics`.

**Comment by neenjaw** [comment]
I think your option of just having it input from as a float rather than a string is probably the most straight forward. We can always write another exercise for splitting strings and extracting numbers.
